lightning/config/config.go

```diff
@@ -286,9 +290,47 @@ func (cfg *Config) Load() error {
 	return nil
 }
 
-func (cfg *Config) Adjust() {
+// Adjust fixes the invalid or unspecified settings to reasonable valid values.
+func (cfg *Config) Adjust() error {
 	cfg.App.Config.Adjust()
 
+	// automatically determine the TiDB port & PD address from TiDB settings
+	if cfg.TiDB.Port <= 0 || len(cfg.TiDB.PdAddr) == 0 {
+		resp, err := http.Get(fmt.Sprintf("http://%s:%d/settings", cfg.TiDB.Host, cfg.TiDB.StatusPort))
+		if err != nil {
+			return errors.Annotate(err, "cannot fetch settings from TiDB, please manually fill in `tidb.port` and `tidb.pd-addr`")
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			return errors.Errorf("TiDB settings returned %s, please manually fill in `tidb.port` and `tidb.pd-addr`", resp.Status)
+		}
+		var settings struct {
```

Comment:
use `tidb/config.Config` directly, even it contains too many other fields whose we don't need